winpr/libwinpr/sysinfo/sysinfo.c

```diff
@@ -553,35 +544,78 @@ DWORD GetTickCount(void)
 
 ULONGLONG winpr_GetTickCount64(void)
 {
-	ULONGLONG ticks = 0;
-#if defined(__linux__)
-	struct timespec ts;
+	const UINT64 ns = winpr_GetTickCount64NS();
+	return WINPR_TIME_NS_TO_MS(ns);
+}
 
-	if (!clock_gettime(CLOCK_MONOTONIC_RAW, &ts))
-		ticks = (ts.tv_sec * 1000) + (ts.tv_nsec / 1000000);
+#endif
+
+UINT64 winpr_GetTickCount64NS(void)
+{
+	UINT64 ticks = 0;
+#if _POSIX_C_SOURCE >= 200112L
+	struct timespec ts = { 0 };
+
+	if (clock_gettime(CLOCK_MONOTONIC_RAW, &ts) == 0)
+		ticks = (ts.tv_sec * 1000000000ull) + ts.tv_nsec;
 
 #elif defined(_WIN32)
-	FILETIME ft;
-	ULARGE_INTEGER ul;
-	GetSystemTimeAsFileTime(&ft);
-	ul.LowPart = ft.dwLowDateTime;
-	ul.HighPart = ft.dwHighDateTime;
-	ticks = ul.QuadPart;
+	union
+	{
+		FILETIME ft;
+		ULARGE_INTEGER ul;
```

Comment:
Microsoft actually recommends using native 64-bit integer types instead of their `LARGE_INTEGER` structs: https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-large_integer-r1